libs/blocks/global-navigation/blocks/profile/signIn.js

```diff
@@ -0,0 +1,30 @@
+const decorateSignIn = async ({ blockEl, decoratedEl, toFragment, getPlaceholder }) => {
+  const dropDown = blockEl.querySelector(':scope > div:nth-child(2)');
+  const signInLabel = await getPlaceholder('sign-in').then(({ value }) => value);
+  if (!dropDown) {
+    const signIn = toFragment`<a href="#" daa-ll="${signInLabel}" class="feds-signin">${signInLabel}</a>`;
+    signIn.addEventListener('click', (e) => {
+      e.preventDefault();
+      window.adobeIMS.signIn();
+    });
+    return signIn;
+  }
+
+  const signInWithDropDown = toFragment`<a href="#" daa-ll="${signInLabel}" class="feds-signin" role="button" aria-expanded="false" aria-controls="navmenu-profile" daa-lh="header|Close" aria-haspopup="true">${signInLabel}</a>`;
+  dropDown.classList.add('feds-signin-dropdown');
+  decoratedEl.insertAdjacentElement('beforeend', dropDown);
+
+  // TODO we don't have a good way of adding attributes to links
+  const dropDownSignIn = dropDown.querySelector('[href="https://adobe.com?sign-in=true"]');
+  if (dropDownSignIn) {
+    dropDownSignIn.addEventListener('click', (e) => {
+      e.preventDefault();
+      window.adobeIMS.signIn();
```

Comment:
Since the profile block might eventually be exposed to non-Milo consumers, maybe it makes sense to check that `adobeIMS.signIn` exists as a function before calling it?

Reply:
So, there's a few different scenarios
1. Standalone Profile - In the old beagle-based profile, we'd only ship the profile itself - consumers cared for sign in themselves. (We can think about offering that too)
2. Standalone Nav - We strictly control IMS/Sign in flow, so it's available
3. Usual milo nav usecase - We strictly control IMS/Sign in flow, so it's available

So I'd like to keep it as is, just so that lana picks up any JS errors that we can monitor and fix (especially, but not only - when we launch)

Reply:
There's one more use-case: imslib is loaded through npm and `adobeIMS` object is not available. In that case we might need to trigger some events to get the lib instance. Not exactly something we want to focus on right now, but add a TODO

Reply:
Added a todo